[functionalize] Avoid one more call to custom get_device on FunctionalTensorWrapper

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #163019
* #162987

Trying to reduce the number of `__torch_dispatch__` calls of FakeTensorMode in the AOT metadata collection pass.

anijain2305 committed:
[functionalize] Avoid one more call to custom get_device on FunctionalTensorWrapper

[ghstack-poisoned]

Lucaskabela reviewed (approved):
LGTM :)

anijain2305 committed:
Update on "[functionalize] Avoid one more call to custom get_device on FunctionalTensorWrapper"


Trying to reduce the number of `__torch_dispatch__` calls of FakeTensorMode in the AOT metadata collection pass.

[ghstack-poisoned]

bdhirsh reviewed (approved):
sgtm if this passes CI

anijain2305 committed:
Update on "[functionalize] Avoid one more call to custom get_device on FunctionalTensorWrapper"


Trying to reduce the number of `__torch_dispatch__` calls of FakeTensorMode in the AOT metadata collection pass.

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge
